WordPress/src/main/java/org/wordpress/android/viewmodel/domainregister/DomainSuggestionsViewModel.kt

```diff
@@ -0,0 +1,130 @@
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.SiteActionBuilder
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.network.rest.wpcom.site.DomainSuggestionResponse
+import org.wordpress.android.fluxc.store.SiteStore.OnSuggestedDomains
+import org.wordpress.android.fluxc.store.SiteStore.SuggestDomainsPayload
+import org.wordpress.android.models.networkresource.ListState
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+import java.util.ArrayList
+import javax.inject.Inject
+import kotlin.properties.Delegates
+
+typealias DomainSuggestionsListState = ListState<DomainSuggestionResponse>
+
+class DomainSuggestionsViewModel @Inject constructor(
+    private val dispatcher: Dispatcher
+) : ViewModel() {
+    lateinit var site: SiteModel
+    private var isStarted = false
+    private val handler = Handler()
+
+    private val _suggestions = MutableLiveData<DomainSuggestionsListState>()
+    val suggestionsLiveData: LiveData<DomainSuggestionsListState>
+        get() = _suggestions
+
+    private var suggestions: ListState<DomainSuggestionResponse>
+            by Delegates.observable(ListState.Init()) { _, _, new ->
+                _suggestions.postValue(new)
+            }
+
+    private val _selectSuggestion = MutableLiveData<DomainSuggestionResponse?>()
+    val selectedSuggestion: LiveData<DomainSuggestionResponse?>
+        get() = _selectSuggestion
+
+    private val _selectedPosition = MutableLiveData<Int>()
+    val selectedPosition: LiveData<Int>
+        get() = _selectedPosition
+
+    var searchQuery: String by Delegates.observable("") { _, oldValue, newValue ->
+        if (newValue != oldValue) {
+            if (!TextUtils.isEmpty(newValue)) {
+                submitSearch(newValue, true)
+            } else {
+                initializeDefaultSuggestions()
+            }
+        }
+    }
+
+    // Bind Dispatcher to Lifecycle
+
+    init {
+        dispatcher.register(this)
+    }
+
+    override fun onCleared() {
+        dispatcher.unregister(this)
+        super.onCleared()
+    }
+
+    fun start(site: SiteModel) {
+        if (isStarted) {
+            return
+        }
+        this.site = site
+        isStarted = true
+        initializeDefaultSuggestions()
+    }
+
+    private fun initializeDefaultSuggestions() {
+        searchQuery = site.name
+    }
+
+    private fun submitSearch(query: String, delayed: Boolean) {
+        if (delayed) {
+            handler.postDelayed({
+                if (query == searchQuery) {
+                    submitSearch(query, false)
+                }
+            }, 250)
+        } else {
+            suggestions = ListState.Ready(ArrayList())
+            fetchSuggestions()
+        }
+    }
+
+    // Network Request
+
+    private fun fetchSuggestions() {
+        // Disable Load more
+        suggestions = ListState.Loading(suggestions, false)
```

Comment:
Do we need to explicitly set `canLoadMore` attribute to false? False is a default value, isn't it?